client/blocks/site-address-changer/index.jsx

```diff
@@ -71,8 +77,23 @@ export class SiteAddressChanger extends Component {
 			domainFieldValue,
 			newDomainSuffix.substr( 1 ),
 			oldDomain,
-			type
+			type,
+			option === 'discard',
+			option !== 'redirect'
 		);
+
+		if ( option === 'redirect' ) {
+			this.props.shoppingCartManager
+				.addProductsToCart( [
+					fillInSingleCartItemAttributes(
+						siteRedirect( { domainFieldValue } ),
```

Comment:
I was not able to complete the checkout, an error was returned from the API:

![image](https://user-images.githubusercontent.com/18705930/132251330-9073d15e-3232-4051-a910-67160943d87d.png)

Perhaps because the properties here are wrong? I don't see `domainFieldValue` as a property of `siteRedirect`. I think it should be `domain`: 
```suggestion
						siteRedirect( { domain: domainFieldValue } ),
```